tests/e2e/dynamic_provisioning.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/tests/e2e/driver"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/tests/e2e/testsuites"
+)
+
+var _ = Describe("[ebs-csi] Dynamic Provisioning", func() {
+	f := framework.NewDefaultFramework("ebs")
+
+	var (
+		cs        clientset.Interface
+		ns        *v1.Namespace
+		ebsDriver driver.DynamicPVTestDriver
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		ebsDriver = driver.InitEbsCSIDriver()
+	})
+
+	for _, t := range driver.ValidVolumeTypes {
+		for _, fs := range driver.ValidFSTypes {
+			volumeType := t
+			fsType := fs
+			Context(fmt.Sprintf("with %q volumeType and %q fsType", volumeType, fsType), func() {
+				It("should create a volume on demand", func() {
+					pods := []testsuites.PodDetails{
+						{
+							Cmd: "echo 'hello world' > /mnt/test-1/data && grep 'hello world' /mnt/test-1/data",
+							Volumes: []testsuites.VolumeDetails{
+								{
+									VolumeType: volumeType,
+									FSType:     fsType,
+									ClaimSize:  driver.MinimumSizeForVolumeType(volumeType),
+									VolumeMount: testsuites.VolumeMountDetails{
+										NameGenerate:      "test-volume-",
+										MountPathGenerate: "/mnt/test-",
+									},
+								},
+							},
+						},
+					}
+					test := testsuites.PodDynamicVolumeWriterReaderTest{
+						CSIDriver: ebsDriver,
+						Pods:      pods,
+					}
+					test.Run(cs, ns)
+				})
+			})
+		}
+	}
+
+	It("should create multiple PV objects, bind to PVCs and attach all to a single pod", func() {
+		volumeType1 := "gp2"
+		volumeType2 := "io1"
+		pods := []testsuites.PodDetails{
+			{
+				Cmd: "echo 'hello world' > /mnt/test-1/data && echo 'hello world' > /mnt/test-2/data && grep 'hello world' /mnt/test-1/data  && grep 'hello world' /mnt/test-2/data",
```

Comment:
This is not a big deal for now. Seems the mount path in `Cmd` is depending on how `tpod.SetupVolume` works on generating the mount path and volume name. This might cause some issue when generation logic is changed but `Cmd` is not updated to reflect the change.

Reply:
Yeah I didnt like that either :(, as the tests evolve and more are written I'll try to find a way to improve that.